src/ui/ResultLink/ResultLink.ts

```diff
@@ -359,15 +362,19 @@ export class ResultLink extends Component {
     if (!this.options.titleTemplate) {
       return this.result.title
         ? HighlightUtils.highlightString(this.result.title, this.result.titleHighlights, null, 'coveo-highlight')
-        : this.result.clickUri;
+        : this.encodedClickUri;
     } else {
       let newTitle = StringUtils.buildStringTemplateFromResult(this.options.titleTemplate, this.result);
       return newTitle
         ? StreamHighlightUtils.highlightStreamText(newTitle, this.result.termsToHighlight, this.result.phrasesToHighlight)
-        : this.result.clickUri;
+        : this.encodedClickUri;
     }
   }
 
+  private get encodedClickUri() {
+    return encodeURI(this.result.clickUri);
```

Comment:
We don't want to `encodeURI` here (encode so it's safely usable in an URL), we want to `escape` for HTML injection. 

For this you should use _.escape